### PR TITLE
Implement actual code to send slack notifications

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR implements codes to actually send slack notifications. The main steps are:
* creating a client using token provided
* do a linear search based on channel names
* send message to each channel

## Related issue number (if any)
ENG-2267

## Loom demo (if any)
First part of https://www.loom.com/share/b7929818b83743d881db8d712675fdc3

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/notification/slack.go [hunk 1]
```diff
@@ -26,13 +29,86 @@ func (s *SlackNotification) Level() shared.NotificationLevel {
...
+	for _, name := range names {
+		namesSet[name] = true
+	}
+
+	// Slack channel names should be unique, based on
+	// https://twitter.com/slackhq/status/965358920146608128?lang=en .
+	// However, we make this list dynamic size on purpose to prevent potential
+	// compatibility issue in future. We will still send notifications to all
+	// channels matching the given name.
+	results := []slack.Channel{}
```
Comment (kenxu95) [inline src/golang/lib/notification/slack.go:54]: I think we should be using `make(slack.Channel, 0, len(channels))` here instead. It's the same thing, but more canonical and makes the append more efficient.

Comment (kenxu95) [reply src/golang/lib/notification/slack.go:54]: I guess I'm not entirely sure what you mean by "making the list of dynamic size on purpose".

**Review by kenxu95 — CHANGES_REQUESTED** [review]
Nice! Some small requests and then will stamp.

File: src/golang/lib/notification/slack.go (hunk 1 above)
Comment (likawind) [reply src/golang/lib/notification/slack.go:54]: I thought `make()` will put a hard-limit on the capacity but it appears the size still extend if `append(` go beyonds the original capacity. So it should actually work!

Comment (kenxu95) [reply src/golang/lib/notification/slack.go]: yeah it's the same thing but generally better since you can specify the capacity for it to pre-allocate.

**Review by kenxu95 — APPROVED** [review]

**Pull request merged**